### ISSUE TITLE
machine.ADC.read_timed(buf, timer) missing

### ISSUE THREAD

**Issue by TDStrasser** [kookaberry/kooka-releases#9]
In release 1.10.0 the machine.ADC.read_timed(buf, timer) function is missing.  Is this an oversight or deliberate?
Does it require a hardware Timer or will a soft Timer do the job (given the sample rate will be relatively high).
Without this function support for sound spectrum and tone detection will no longer work (per the Goertzel algorithm used thus far).
Regards,
Tony

**Comment by TDStrasser** [comment]
My mistake - I found it in the kooka module - forgotten it isn't part of machine.ADC.

**Comment by dpgeorge** [comment]
I don't think this is implemented for the RP2040.  Which firmware are you using?

**Comment by TDStrasser** [comment]
It's in the STM32 firmware but not on RP2040. Both using the latest release 1.10.0.
Logically it should be a machine.ADC method on both.

**Comment by TDStrasser** [comment]
Hi Damien, just a bump on this issue as a firmware update is pending, as I understand according to Julian.  We do need the ADC read_timed functionality (eg. for sound sampling and characterisation) and it should be consistent on both processors. Thanks, Tony.

**Comment by dpgeorge** [comment]
@TDStrasser I have a `read_timed()` function working on the RP2040/RP2350.  I put it in the `kooka` module as `kooka.read_timed(adc, buf, freq)` to match the STM32 firmware.

If you need, I can put it in `machine.ADC.read_timed(buf, freq)`.  But only on the RP2040/RP2350.

I'd prefer to keep it as `kooka.read_timed` so it's consistent on both.

Also, what kind of frequencies do you use this function with?

**Comment by TDStrasser** [comment]
Thank you Damien.  In the past we have used it (for bees buzzes) at 2000Hz sample rate (i.e. 2 x the highest expected frequency of 1000Hz).  As it will mostly be used for audio sampling, say at telephony bandwidth, then one could conceive of using up to 6.6kHz sampling rate (if the ADC is capable of that).
We can live with the kooka module function as for the STM32 firmware for the sake of consistency.
Cheers, Tony.

**Comment by dpgeorge** [comment]
OK, thanks for the info.  I tested it at 10kHz and it worked well.

**Comment by TDStrasser** [comment]
Thanks again - 10kHz should be plenty for our purposes.

**Comment by dpgeorge** [comment]
`kooka.read_timed()` was added in v1.11.0.
